### PR TITLE
Initializer Contracts with arguments

### PR DESCRIPTION
**Pull request by frangio**
Explanation included in README.

There's a couple of approaches that I haven't implemented yet.

- [x] Deploy a new contract for each instantiation
- [ ] Replace `codecopy` opcodes by `calldatacopy`
- [ ] Place arguments in memory and remove/jump over the part that calls `codecopy`

### CODE REVIEW

File: initializer_contracts_with_args/scripts/poc-2.js [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+    } else {
+      await sleep(100);
+    }
+  }
+}
+
+async function deploy(bytecode) {
+  const tx = await pweb3.eth.sendTransaction({
+    from: (await pweb3.eth.getAccounts())[0],
+    to: 0,
```
Comment (spalladino) [inline initializer_contracts_with_args/scripts/poc-2.js]: Should be `null` if we want this to run in geth/parity

File: initializer_contracts_with_args/scripts/poc-2.js [hunk 2]
```diff
@@ -0,0 +1,76 @@
+function sleep(ms) {
+  return new Promise(resolve => setTimeout(resolve, ms));
+}
+
+async function getTransactionReceipt(txid) {
+  while (true) {
+    const receipt = await pweb3.eth.getTransactionReceipt(txid);
```
Comment (spalladino) [inline initializer_contracts_with_args/scripts/poc-2.js]: I take it pweb3 is injected by buidler. Nice addition.

File: initializer_contracts_with_args/scripts/poc-2.js [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+// push1 00  | 0x60 0x00  | 0xIS 0x00
+// return    | 0xf3
+const ASM_RETURN_REST = '600d80380380916000396000f3';
+
+function prepareConstructor(bytecode) {
+  return bytecode.replace('0x', `0x${ASM_RETURN_REST}${ASM_RETURN_REST}`);
+}
+
+function getArgs(contract, args) {
+  return web3.eth.contract(contract.abi).new.getData(...args, { data: '0x' });
```
Comment (spalladino) [inline initializer_contracts_with_args/scripts/poc-2.js]: Very clever!

**Comment by frangio** [comment]
I'm merging this since we've decided to wait a bit before working on the other alternatives.

**Pull request merged**